specification/search/Azure.Search/models-index.tsp

```diff
@@ -0,0 +1,1077 @@
+   * Relative weight of the vector query when compared to other vector query and/or
+   * the text query within the same search request. This value is used when
+   * combining the results of multiple ranking lists produced by the different
+   * vector queries and/or the results retrieved through the text query. The higher
+   * the weight, the higher the documents that matched that query will be in the
+   * final ranking. Default is 1.0 and the value needs to be a positive number
+   * larger than zero.
+   */
+  weight?: float32;
+}
+
+/**
+ * Contains a document found by a search query, plus associated metadata.
+ */
+#suppress "@azure-tools/typespec-azure-core/bad-record-type" "model should have additionalProperties"
+model SearchResult {
+  ...Record<unknown>;
+
+  /**
+   * The relevance score of the document compared to other documents returned by the
+   * query.
+   */
+  @visibility("read")
+  @encodedName("application/json", "@search.score")
+  score: float64;
+
+  /**
+   * The relevance score computed by the semantic ranker for the top search results.
+   * Search results are sorted by the RerankerScore first and then by the Score.
+   * RerankerScore is only returned for queries of type 'semantic'.
+   */
+  @visibility("read")
+  @encodedName("application/json", "@search.rerankerScore")
+  rerankerScore?: float64;
+
+  /**
+   * Text fragments from the document that indicate the matching search terms,
+   * organized by each applicable field; null if hit highlighting was not enabled
+   * for the query.
+   */
+  @visibility("read")
+  @encodedName("application/json", "@search.highlights")
+  highlights?: Record<string[]>;
+
+  /**
+   * Captions are the most representative passages from the document relatively to
+   * the search query. They are often used as document summary. Captions are only
+   * returned for queries of type 'semantic'.
+   */
+  @visibility("read")
+  @encodedName("application/json", "@search.captions")
+  captions?: QueryCaptionResult[];
+}
+
+/**
+ * Captions are the most representative passages from the document relatively to
+ * the search query. They are often used as document summary. Captions are only
+ * returned for queries of type `semantic`.
+ */
+#suppress "@azure-tools/typespec-azure-core/bad-record-type" "model should have additionalProperties"
+model QueryCaptionResult {
+  ...Record<unknown>;
+
+  /**
+   * A representative text passage extracted from the document most relevant to the
+   * search query.
+   */
+  @visibility("read")
+  text?: string;
+
+  /**
+   * Same text passage as in the Text property with highlighted phrases most
+   * relevant to the query.
+   */
+  @visibility("read")
+  highlights?: string;
+}
+
+/**
+ * Response containing suggestion query results from an index.
+ */
+model SuggestDocumentsResult {
+  /**
+   * The sequence of results returned by the query.
+   */
+  @visibility("read")
+  @encodedName("application/json", "value")
+  results: SuggestResult[];
```

Comment:
Why isnt this just named value?

Reply:
Is this because of a sdk renaming? If it's for client libraries, then we should do those renamings in client.tsp and have the service's actual property names in the main typespec files. 

Except for those that are in the shape of `@search.x` those we can keep specifying with the @encodedName decorator